lib/puppetserver/ca/action/sign.rb

```diff
@@ -74,45 +73,12 @@ def run(input)
           return success ? 0 : 1
         end
 
-        def http_client(settings)
-          @client ||= HttpClient.new(settings)
-        end
-
-        def get_certificate_statuses(settings)
-          client = http_client(settings)
-          url = client.make_ca_url(settings[:ca_server],
-                                   settings[:ca_port],
-                                   'certificate_statuses',
-                                   'any_key')
-          client.with_connection(url) do |connection|
-            connection.get(url)
-          end
-        end
-
-        def sign_certs(certnames,settings)
-          results = {}
-          client = http_client(settings)
-          url = client.make_ca_url(settings[:ca_server],
-                                   settings[:ca_port],
-                                   'certificate_status')
-          client.with_connection(url) do |connection|
-            certnames.each do |certname|
-              url.resource_name = certname
-              results[certname] = connection.put(BODY, url)
-            end
-          end
-          return results
+        def ca(settings)
+          @ca ||= Puppetserver::Ca::CertificateAuthority.new(@logger, settings)
         end
 
         def get_all_certs(settings)
-          result = get_certificate_statuses(settings)
-
-          unless result.code == '200'
-              @logger.err 'Error:'
-              @logger.err "    #{result.inspect}"
-              return nil
-          end
-          return result
+          ca(settings).get_certificate_statuses
```

Comment:
Since there's not much in here any more should we just fold these up into `#run` or a method that run calls?